ethereum/datastructures/src/main/java/tech/pegasys/artemis/datastructures/util/BeaconStateUtil.java

```diff
@@ -220,6 +152,10 @@ private static void process_deposit(
                     compute_domain(DOMAIN_DEPOSIT));
         if (!proof_is_valid) {
           STDOUT.log(Level.DEBUG, "Skipping invalid deposit");
+          if (pubKeyToIndexMap != null) {
+            // The validator won't be created so the calculated index won't be correct
+            pubKeyToIndexMap.remove(pubkey);
+          }
```

Comment:
We were incorrectly storing the validator index even if the deposit was invalid. We would have incremented the wrong balance if there was a later, valid deposit for that validator as it would have used the cached index from the invalid deposit even though the validator didn't actually get created then.